dockerfiles/ink-waterfall-ci-linux/Dockerfile

```diff
@@ -0,0 +1,60 @@
+ARG VCS_REF=master
+ARG BUILD_DATE
+ARG REGISTRY_PATH=paritytech
+
+FROM ${REGISTRY_PATH}/base-ci:latest
+
+# metadata
+LABEL io.parity.image.authors="devops-team@parity.io" \
+	io.parity.image.vendor="Parity Technologies" \
+	io.parity.image.title="${REGISTRY_PATH}/ink-waterfall-ci-linux" \
+	io.parity.image.description="Inherits from paritytech/base-ci. \
+rust nightly, clippy, rustfmt, rust-src, cargo-contract, canvas-node, binaryen" \
+	io.parity.image.source="https://github.com/paritytech/scripts/blob/${VCS_REF}/\
+dockerfiles/ink-waterfall-ci-linux/Dockerfile" \
+	io.parity.image.documentation="https://github.com/paritytech/scripts/blob/${VCS_REF}/\
+dockerfiles/ink-waterfall-ci-linux/README.md" \
+	io.parity.image.revision="${VCS_REF}" \
+	io.parity.image.created="${BUILD_DATE}"
+
+WORKDIR /builds
+
+ENV SHELL /bin/bash
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN	set -eux; \
+	apt-get -y update && \
+	apt-get install -y --no-install-recommends \
+	binaryen && \
+# The supported Rust nightly version must support the following components
+# to allow for a functioning CI pipeline:
+#
+#  - cargo: General build tool.
+#  - rustfmt: Rust formatting tool.
+#  - clippy: Rust linter.
+#
+# We also need to install the wasm32-unknown-unknown target to test
+# ink! smart contracts compilation for this target architecture.
+#
+# Only Rust nightly builds supporting all of the above mentioned components
+# and targets can be used for this docker image.
+# Installs the latest common nightly for the listed components,
+# adds those components, wasm target and sets the profile to minimal
+	rustup toolchain install nightly --target wasm32-unknown-unknown \
+		--profile minimal --component rustfmt clippy rust-src && \
+	rustup default nightly && \
+	cargo install --git https://github.com/paritytech/cargo-contract && \
+	cargo install canvas-node --git https://github.com/paritytech/canvas-node.git --locked && \
+	git clone --depth 1 https://github.com/paritytech/ink.git . && \
```

Comment:
why do you need an ink repo here? This stuff is usually done within CI, there at least you'll get the most recent version.

Reply:
I need it to get to the `examples/`, the idea is to deploy those to the `canvas-node` using `cargo-contract`.